clippy_lints/src/methods/mod.rs

```diff
@@ -2908,18 +2932,85 @@ fn lint_skip_while_next<'tcx>(
     }
 }
 
-/// lint use of `filter().map()` for `Iterators`
+fn is_method<'tcx>(cx: &LateContext<'tcx>, expr: &hir::Expr<'_>, method_path: &[&str], method_name: Symbol) -> bool {
```

Comment:
Something similar is also done in `bind_instead_of_map` and other modules. In the `Closure` case you should call 

```rust
let closure_body = cx.tcx.hir().body(body_id);
let closure_expr = remove_blocks(&closure_body.value);
```

and then just match the `closure_expr` against `ExprKind::MethodCall`. On `args[0]` you can just check with `is_type_diagnostic_item`.

Reply:
I think my [previous comment](https://github.com/rust-lang/rust-clippy/pull/6342/files#r545862641) still applies here. This closure unpacking now appears in 4 lints across 4 files. This should probably be moved to the methods/utils module. But this should be done in another PR.